config/secrets/transform.go

```diff
@@ -38,57 +40,32 @@ func (provider) Transform(config map[string]interface{}) error {
 	s := secrets.New(creds)
 
 	// Overwrite the baseUrl for secrets if one is given
-	if baseURL, _ := config["secretsBaseUrl"].(string); baseURL != "" {
+	if baseURL, _ := cfg["secretsBaseUrl"].(string); baseURL != "" {
 		s.BaseURL = baseURL
 	}
 
 	// Create a cache to avoid loading the same secret twice, we use the same
 	// creds for all calls and we don't persistent the cache so there is no risk
 	// of scope elevation here.
-	cache := map[string]map[string]interface{}{}
+	cache := make(map[string]map[string]interface{})
 
-	_, err := injectSecrets(s, cache, config)
-	return err
-}
-
-func injectSecrets(
-	s *secrets.Secrets,
-	cache map[string]map[string]interface{},
-	val interface{},
-) (interface{}, error) {
-	switch val := val.(type) {
-	case []interface{}:
-		for i, v := range val {
-			v, err := injectSecrets(s, cache, v)
-			val[i] = v
-			if err != nil {
-				return nil, err
-			}
-		}
-	case map[string]interface{}:
-		name, ok1 := val["$secret"].(string)
-		key, ok2 := val["key"].(string)
-		if ok1 && ok2 && len(val) == 2 {
-			// If secret isn't in the cache we try to load it
-			if _, ok := cache[name]; !ok {
-				secret, err := s.Get(name)
-				if err != nil {
-					return nil, err
-				}
-				value := map[string]interface{}{}
-				_ = json.Unmarshal(secret.Secret, &value)
```

Comment:
We should probably `nilOrPanic(...)` rather than ignore errors - what do you think? I think we should definitely have something to recover from panics at the bottom of the stack, which can initiate a worker shutdown. It feels non-deterministic to ignore errors, even if they should never occur... I also don't like bloating code with error handling, so it is definitely a balance.